### DISCUSSION TITLE
Utilização de Listas em arquivos no Terraform

### DISCUSSION THREAD

**Discussion by debroski** [chgasparoto/curso-aws-com-terraform#91] [Geral]
Boa tarde Pessoal,

Estou em duvida como utilizar listas no terraform (não sei ainda se isso é possivel)

Preciso criar uma sequencia e politicas para buckets no s3, onde deve existir uma politica para cada bucket.
É possivel criar uma lista.txt com os nomes dos buckets e executar como um "for" no terraform para ir lendo esta lista? 
Se sim alguem sabe como eu trabalho?

**Comment by chgasparoto** [comment]
@debroski sim, é possível fazer isso.

Você pode usar a função [file](https://www.terraform.io/language/functions/file) para ler o arquivo e a função [split](https://www.terraform.io/language/functions/split) para transformar a string, retornada da função anterior, para uma lista. Com a lista em mãos você consegue iterar sobre ela normalmente.
